### ISSUE TITLE
Actor Critic Bipedal Walker

### ISSUE THREAD

**Issue by RossMelbourne** [FitMachineLearning/FitML#1]
First of all, thank you for posting really interesting machine learning code.  I am still trying to learn reinforcement learning and I have run your BepedalWalker_A2C.py. Even with running 10,000 games the walker still does not truly walk and just drags one knee behind itself.  Is this what you experienced?  What adjustments do you recommend I make to the hyper parameters to try?  Is any of this dependent upon the CPU/CPU and version of Tensorflow or Keras?

**Comment by FitMachineLearning** [comment]
That is my experience also.
The are a few issues here.

1) the sample weight param for Keras fit doesn't seem to work as advertised.
2) without it one would have to write their custom cost function, which I haven't completed yet.
3) I haven't tried with a very large memory yet. this would enable the agent to learn and retain more situations.

I just haven't had a chance to try all of these yet. Probably when I am back from vacation.

**Comment by FitMachineLearning** [comment]
Note that I have stopped research on the policy gradient to focus my new alternate approach to continuous action space control problems.

I am calling it "Selective Memory" and you can see it in my repository.

**Comment by RossMelbourne** [comment]
I will try out your Selective Memory script and see what results I get.  I just noticed you posted a question on OpenAI about a free continuous action space environment.  I have just started looking at Unity for this.  I have done Unity development in the past so it is somewhat familiar to me. 
 
https://blogs.unity3d.com/2017/09/19/introducing-unity-machine-learning-agents/ 

Unity is interesting to me because I need a photo realistic simulation environment to train a robot to navigate in the real world.

**Comment by FitMachineLearning** [comment]
Thanks for the recommendation Ross.
Unity looks very interesting. If it works on Python without too much fidgeting, then I can directly apply my agents. Looking forward to trying it out this weekend.

BTW. I have completed phase 1 research on "Selective Memory". it is now working and can be found here
https://github.com/FitMachineLearning/FitML/tree/master/SelectiveMemory

I recommend you use version 5. You should start to see it run after 3000 iterations of training.

**Comment by RossMelbourne** [comment]
I have been running Bipedal_Walker_v5.py on two difference machines (Mac OS and Linux) and after 3,000 iterations it is up on two feet but it is still not really walking.  Did you change the file at all after posting V5?

**Comment by FitMachineLearning** [comment]
Hi Ross,

I reran BipedalWalker_v5.py on both Windows and Mac and you can see the agent leap-walking even after 500 iterations. I have made a video of it on youtube and will share the link as soon as I am back at my desk.

Please note that the agent (depending on hyperparameters) disregards the notion of running as we know it for a safer gallop mode of locomotion.

**Comment by FitMachineLearning** [comment]
Link to video as promised
https://youtu.be/hKrFFeZqq3E
